src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -414,21 +414,43 @@ void CoreWorkerDirectTaskReceiver::HandlePushTask(
     send_reply_callback(Status::Invalid("client cancelled stale rpc"), nullptr, nullptr);
   };
 
-  auto it = scheduling_queue_.find(task_spec.CallerWorkerId());
-  if (it == scheduling_queue_.end()) {
-    auto result = scheduling_queue_.emplace(
-        task_spec.CallerWorkerId(),
-        SchedulingQueue(task_main_io_service_, *waiter_, worker_context_));
-    it = result.first;
-  }
+
   auto dependencies = task_spec.GetDependencies();
-  // Pop the dummy actor dependency.
+
   if (task_spec.IsActorTask()) {
+    auto it = actor_scheduling_queues_.find(task_spec.CallerWorkerId());
+    if (it == actor_scheduling_queues_.end()) {
+      auto result = actor_scheduling_queues_.emplace(
+          task_spec.CallerWorkerId(),
+          new ActorSchedulingQueue(task_main_io_service_, *waiter_, worker_context_));
+      it = result.first;
+    }
+    
+    // Pop the dummy actor dependency.
     // TODO(swang): Remove this with legacy raylet code.
     dependencies.pop_back();
+    it->second->Add(request.sequence_number(), request.client_processed_up_to(),
+                   accept_callback, reject_callback, dependencies);
+  } else {
+
+    // Add the normal task's callbacks to the non-actor scheduling queue.
+    normal_scheduling_queue_->Add(request.sequence_number(),
+                                 request.client_processed_up_to(), accept_callback,
+                                 reject_callback, dependencies);
+  }
+}
+
+
+
+void CoreWorkerDirectTaskReceiver::RunNormalTasksFromQueue() {
+  // If the scheduling queue is empty, return.
+  if (normal_scheduling_queue_->TaskQueueEmpty()) {
```

Comment:
I think we need to hold a lock here because `normal_scheduling_queue_` will now get accessed by both the task execution event loop and the IO event loop.